File: StudyBuddy.Model/LogMessage.cs
using System;

namespace StudyBuddy.Model
{
    public class LogMessage
    {
        public DateTime Occurence { get; set; } = DateTime.Now;
        public string Message { get; set; }
        public LogLevel Level { get; set; }
        public int UserId { get; set; }
        public Component Source { get; set; }
    }
}

File: StudyBuddy.Backend/src/app/model/logmessage.ts
export class LogMessage {
    occurence:string = "";
    message:string = "";
    level:number = 0;
    userId:number = 0;
    source:number = 0;

    constructor() { }

    static fromApi(obj):LogMessage {
        let result = new LogMessage();
        result.occurence = obj['occurence'];
        result.message = obj['message'];
        result.level = +obj["level"];
        result.userId = +obj["userId"];
        result.source = +obj["source"];
        return result;
    }

    toApi() {
        return { 
            'Message': this.message,
            'Level': this.level,
            'UserId': this.userId,
            'Source': this.source
        };
    }
}